### ISSUE TITLE
Streamed Response seems not to work always

### ISSUE THREAD

**Issue by Scyks** [playframework/playframework#6621]
### Play Version (2.5.x / etc)

2.5.3
### API (Scala / Java / Neither / Both)

Scala 2.11.8
### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

Linux lamp 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-2+deb8u3 (2016-07-02) x86_64 GNU/Linux.
### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode)
### Expected Behavior

Create a file on demand and send it directly to the user using output streams.
1. create a stream as output to create a file on demand and stream this to the user
2. send it to the user
3. it works all the time
### Actual Behavior

We figured out a behavior in which this creation works 7 times and at the 8th time there is no response from play anymore at all. Not only the export stops, also everything else does not work anymore. Only a restart helps.
1. create a stream
2. write into stream
3. 7 times the user receives the file
4. 8th time - no response
### Reproducible Test Case

The code we used is something like that:

``` scala
val (outputStream, publisher) = StreamConverters.asOutputStream().toMat(Sink.asPublisher(false))(Keep.both).run()
val res = HttpEntity.Streamed(Source.fromPublisher(publisher), None, Some("plain/text"))
Future {
    outputStream.write("hello world".getBytes())
    outputStream.flush()
    outputStream.close()
}
Ok.sendEntity(res).withHeaders("Content-Disposition" -> "attachment; filename=file.txt;")
```

The working version is something like this:

``` scala
val in = new PipedInputStream
val out = new PipedOutputStream(in)
val source = StreamConverters.fromInputStream(() => in)
Future {
    out.write("hello world".getBytes())
    out.flush()
    out.close()
}
Ok.sendEntity(HttpEntity.Streamed(source, None, None)).withHeaders("Content-Disposition" -> "attachment; filename=file.txt;")
```

Actually i don't know if this is related to Play or to akka-streams. But maybe you'll find out.

**Comment by schmitch** [comment]
Unfortunatly, this is not a Bug in Play!

I actually made a PR against akka, to fix this, but I'm not sure how fast I can get it trough (it's my first there).

https://github.com/akka/akka/pull/21866

You might checkout the PR and do `publishLocal` akka and test it against the snapshot if you want.

**Comment by schmitch** [comment]
This will be fixed in Akka `2.4.15`. Which is coming soon, so closing.

**Comment by mkurz** [comment]
https://github.com/akka/akka/releases/tag/v2.4.15
